libs/ui/src/react_query.ts

```diff
@@ -1,6 +1,25 @@
 // eslint-disable-next-line vx/gts-no-import-export-type
 import type { DefaultOptions } from '@tanstack/react-query';
 
+/**
+ * A custom retry function
+ *
+ * A 504 in VxSuite usually just indicates that the backend is starting up (or restarting), in
+ * which case we should retry, but still with a bound in case the backend is completely failing to
+ * start up.
+ *
+ * 1 original attempt + 4 retry attempts
+ * = 0 + 1 + 2 + 4 + 8 seconds (because of exponential backoff)
+ * = 15 seconds of total wait time
+ *
+ * Otherwise, don't retry. Because everything is local, we don't expect intermittent errors.
+ */
+export function shouldRetry(failedRetryCount: number, error: unknown): boolean {
+  const isBackendLikelyStartingUp =
+    error instanceof Error && error.message.includes('504');
```

Comment:
I wanted to make this condition a little bit more specific, e.g. by checking for `"ECONNREFUSED"` in the error message, but that piece of the backend error message isn't propagated through the frontend proxy

This feels like a reasonable starting point, though